src/validation.cpp

```diff
@@ -648,19 +656,28 @@ bool CheckTransaction(const CTransaction &tx, CValidationState &state, bool fChe
                 return false;
         }
 
-        if(tx.IsLelantusTransaction()) {
+        if (tx.IsLelantusTransaction()) {
             if (!CheckLelantusTransaction(tx, state, hashTx, isVerifyDB, nHeight, isCheckWallet, fStatefulZerocoinCheck, sigmaTxInfo, lelantusTxInfo))
                 return false;
         }
 
-        if (!CheckZerocoinTransaction(tx, state, Params().GetConsensus(), hashTx, isVerifyDB, nHeight, isCheckWallet, fStatefulZerocoinCheck, zerocoinTxInfo))
-            return false;
+        auto params = ::Params().GetConsensus();
+        if (tx.IsZerocoinSpend() || tx.IsZerocoinMint()) {
+            if (nHeight != INT_MAX && nHeight >= params.nDisableZerocoinStartBlock)
```

Comment:
we don't expect these kinds of transactions in mempool, you can just check if `nHeight >= ...`.

Reply:
done.